reference/configuration/framework.rst

```diff
@@ -169,39 +293,431 @@ see :doc:`/cookbook/request/load_balancer_reverse_proxy`.
+
+csrf_protection
+...............
+
+.. seealso::
+
+    For more information about CSRF protection in forms, see :ref:`forms-csrf`.
+
+.. _reference-form-csrf_protection-enabled:
+
+enabled
+"""""""
+
+**type**: ``boolean`` **default**: ``true``
+
+Whether to enable the CSRF support in forms or not. Setting this option to
+``true`` requires the `secret`_ to be set and `session`_ to be enabled. It
+becomes the service container parameter named ``form.type_extension.csrf.enabled``.
+
+This option will automatically be set to ``true`` when one of the child
+settings is configured.
+
+.. seealso::
+
+    Read more about CSRF in :ref:`the Book <forms-csrf>`.
+
+field_name
+""""""""""
+
+**type**: ``string`` **default**: ``'_token'``
+
+The name of the hidden field containing the CSRF token in the form. It becomes
+the service container parameter named ``form.type_extension.csrf.field_name``.
+
+esi
+~~~
+
+.. seealso::
+
+    You can read more about Edge Side Includes (ESI) in :ref:`edge-side-includes`.
+
+.. _reference-esi-enabled:
+
+enabled
+.......
+
+**type**: ``boolean`` **default**: ``false``
+
+Whether to enable the edge side includes support in the framework.
+
+You can also set ``esi`` to ``true`` to enable it:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/config.yml
+        framework:
+            esi: true
+
+    .. code-block:: xml
+
+        <!-- app/config/config.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony http://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+
+            <framework:config>
+                <esi />
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // app/config/config.php
+        $container->loadFromExtension('framework', array(
+            'esi' => true,
+        ));
+
+fragments
+~~~~~~~~~
+
+.. seealso::
+
+    Learn more about fragments in the
+    :ref:`HTTP Cache article <book-http_cache-fragments>`.
+
+.. _reference-fragments-enabled:
+
+enabled
+.......
+
+**type**: ``boolean`` **default**: ``false``
+
+Whether to enable the fragment listener or not. The fragment listener is used
+to render ESI fragments independently of the rest of the page.
+
+This setting is automatically set to ``true`` when one of the child settings is
+configured.
+
+.. _reference-fragments-path:
+
+path
+....
+
+**type**: ``string`` **default**: ``'/_fragment'``
+
+The path prefix for fragments. The fragment listener will only be executed
+when the request starts with this path. It becomes the service container
+parameter called ``fragment.path``.
```

Comment:
same thing - I'd have to hear an argument in favor of showing the parameter names :)